File: ruby/lib/Sikaru/types/operation_view_method.rb
# frozen_string_literal: true

module Sikaru
  module Types
    module OperationViewMethod
      extend Sikaru::Internal::Types::Enum

      BASH_RUN = "bash.run"
      BASH_START = "bash.start"
      BASH_READ = "bash.read"
      BASH_WAIT = "bash.wait"
      BASH_CANCEL = "bash.cancel"
      WORKSPACE_WRITE_TEXT = "workspace.write_text"
      BASH_WAIT_FOR = "bash.wait_for"
      JOBS_NEXT_COMPLETED = "jobs.next_completed"
    end
  end
end

File: swift/Sources/Schemas/OperationViewMethod.swift
import Foundation

public enum OperationViewMethod: String, Codable, Hashable, CaseIterable, Sendable {
    case bashStart = "bash.start"
    case bashRead = "bash.read"
    case bashWait = "bash.wait"
    case bashCancel = "bash.cancel"
    case workspaceWriteText = "workspace.write_text"
}